### PR TITLE
[doc] fix profiling.rst for the profiling

### LINKED ISSUE #43627: [<Doc>] Missing steps in the profiling doc
### Description

During our There are some missing steps for the profiling doc.
1. We need to set `"export RAY_RAYLET_PERFTOOLS_PROFILER=1` for cpu profile raylet.
2. `make install` is needed to install jemalloc
3. `graphviz` is needed  to generate svg plot

Fix: 
see https://github.com/ray-project/ray/pull/43615

### Link

_No response_

### PR DESCRIPTION
**Pull request by Bye-legumes**
## Why are these changes needed?
1.Currently the profiling docs for CPU is not correct, i.e., we need to add "export RAY_RAYLET_PERFTOOLS_PROFILER=1" so that we can profile on google-pprof.
2. Split clearly the CPU profiling and memory profiling methods.
3. To correct install the jemalloc, ``make install`` is also needed to correctly install jemalloc.
4. To visualized the plot, the ``graphviz`` is also needed otherwise there is a issue for jemalloc which doesn't show this problem clearly.
5. Add how to get the .svg plot for memory profile.

## Related issue number
Closes #43627

## Checks

- [√] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [√] I've run `scripts/format.sh` to lint the changes in this PR.
- [√] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [√] This PR is not tested :(

### CODE REVIEW

**Comment by Bye-legumes** [comment]
@rkooo567 Hi, can you help me to check this PR or any suggestions?

File: doc/source/ray-contribute/profiling.rst [hunk 1]
```diff
@@ -29,23 +29,58 @@ symbolize on Mac OS have failed.
...
-Launching the to-profile binary
--------------------------------
+You may need to install ``graphviz`` for ``pprof`` to generate flame graphs.
 
-If you want to launch Ray in profiling mode, define the following variables:
+.. code-block:: bash
+  
+  sudo apt-get install graphviz
+
+Launching the CPU Profiling and set to-profile binary file
```
Comment (jjyao) [inline doc/source/ray-contribute/profiling.rst:38]: Let's change the sections to 

```
CPU Profiling
-------------

xxxxxx

Visualizing the CPU profile
~~~~~~~~~~~~~~~~~~~~

xxxxxx

Memory Profiling
-----------------

xxxxx

Visualizing the Memory profile
~~~~~~~~~~~~~~~~~~~~

xxxxx
```

?

**Review by jjyao — COMMENTED** [review]
Thanks a lot for the changes!

File: doc/source/ray-contribute/profiling.rst (hunk 1 above)
Comment (Bye-legumes) [reply doc/source/ray-contribute/profiling.rst]: Changed, you can check.

**Review by jjyao — APPROVED** [review]

File: doc/source/ray-contribute/profiling.rst [hunk 2]
```diff
@@ -29,23 +29,59 @@ symbolize on Mac OS have failed.
...
+  export RAY_RAYLET_PERFTOOLS_PROFILER=1
 
 
 The file ``/tmp/pprof.out`` will be empty until you let the binary run the
 target workload for a while and then ``kill`` it via ``ray stop`` or by
-letting the driver exit.
+letting the driver exit. 
+
+Note: Enabling `RAY_RAYLET_PERFTOOLS_PROFILER` allows profiling of the Raylet component. 
+If you wish to profile other modules, you should utilize `RAY_{MODULE}_PERFTOOLS_PROFILER`,
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:55]:
```suggestion
To profile other modules, use `RAY_{MODULE}_PERFTOOLS_PROFILER`,
```

File: doc/source/ray-contribute/profiling.rst [hunk 3]
```diff
@@ -29,23 +29,59 @@ symbolize on Mac OS have failed.
...
-Launching the to-profile binary
--------------------------------
+You may need to install ``graphviz`` for ``pprof`` to generate flame graphs.
 
-If you want to launch Ray in profiling mode, define the following variables:
+.. code-block:: bash
+  
+  sudo apt-get install graphviz
+
+CPU Profiling
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:38]:
```suggestion
CPU profiling
```

File: doc/source/ray-contribute/profiling.rst [hunk 4]
```diff
@@ -29,23 +29,59 @@ symbolize on Mac OS have failed.
...
+  RAYLET=ray/python/ray/core/src/ray/raylet/raylet
+
+  google-pprof -svg $RAYLET /tmp/pprof.out > /tmp/pprof.svg
+  # Then open the .svg file with Chrome.
+
+  # If you realize the call graph is too large, use -focus=<some function> to zoom
+  # into subtrees.
+  google-pprof -focus=epoll_wait -svg $RAYLET /tmp/pprof.out > /tmp/pprof.svg
+
+Here's a snapshot of an example svg output, taken from the official
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:77]:
```suggestion
Below is a snapshot of an example SVG output, from the official
```

File: doc/source/ray-contribute/profiling.rst [hunk 5]
```diff
@@ -29,23 +29,59 @@ symbolize on Mac OS have failed.
...
+documentation:
+
+.. image:: http://goog-perftools.sourceforge.net/doc/pprof-test-big.gif
 
 Memory Profiling
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:82]:
```suggestion
Memory profiling
```

File: doc/source/ray-contribute/profiling.rst [hunk 6]
```diff
@@ -29,23 +29,59 @@ symbolize on Mac OS have failed.
...
 ----------------
+
 If you want to run memory profiling on Ray core components, you can use Jemalloc (https://github.com/jemalloc/jemalloc).
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:85]:
```suggestion
To run memory profiling on Ray core components, use Jemalloc (https://github.com/jemalloc/jemalloc).
```

File: doc/source/ray-contribute/profiling.rst [hunk 7]
```diff
@@ -29,23 +29,59 @@ symbolize on Mac OS have failed.
...
+
+Note: Enabling `RAY_RAYLET_PERFTOOLS_PROFILER` allows profiling of the Raylet component. 
+If you wish to profile other modules, you should utilize `RAY_{MODULE}_PERFTOOLS_PROFILER`,
+where `MODULE` represents the uppercase form of the process type, such as `GCS_SERVER`.
+
+
+Visualizing the CPU profile
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The output of ``pprof`` can be visualized in many ways. Here we output it as a
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:62]:
```suggestion
You can visualize the output of ``pprof`` in different ways. Below, the output is a
```

File: doc/source/ray-contribute/profiling.rst [hunk 8]
```diff
@@ -66,6 +102,8 @@ Users are supposed to provide 3 env vars for memory profiling.
   cd jemalloc-5.2.1 
   ./configure --enable-prof --enable-prof-libunwind 
   make
+  make install
+
 
   # set jemalloc configs through MALLOC_CONF env variable
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:108]:
```suggestion
  # Set jemalloc configs through MALLOC_CONF env variable.
```

File: doc/source/ray-contribute/profiling.rst [hunk 9]
```diff
@@ -66,6 +102,8 @@ Users are supposed to provide 3 env vars for memory profiling.
...
   # read http://jemalloc.net/jemalloc.3.html#opt.lg_prof_interval
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:109]:
```suggestion
  # Read http://jemalloc.net/jemalloc.3.html#opt.lg_prof_interval.
```

File: doc/source/ray-contribute/profiling.rst [hunk 10]
```diff
@@ -79,28 +117,20 @@ Users are supposed to provide 3 env vars for memory profiling.
   # You should be able to see the following logs.
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:117]:
```suggestion
  # You should see the following logs.
```

File: doc/source/ray-contribute/profiling.rst [hunk 11]
```diff
@@ -79,28 +117,20 @@ Users are supposed to provide 3 env vars for memory profiling.
...
   2021-10-20 19:45:08,175	INFO services.py:622 -- Jemalloc profiling will be used for gcs_server. env vars: {'LD_PRELOAD': '/Users/sangbincho/jemalloc-5.2.1/lib/libjemalloc.so', 'MALLOC_CONF': 'prof:true,lg_prof_interval:33,lg_prof_sample:17,prof_final:true,prof_leak:true'}
 
-Visualizing the CPU profile
----------------------------
-
-The output of ``pprof`` can be visualized in many ways. Here we output it as a
-zoomable ``.svg`` image displaying the call graph annotated with hot paths.
+Visualizing the Memory profile
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:120]:
```suggestion
Visualizing the memory profile
```

File: doc/source/ray-contribute/profiling.rst [hunk 12]
```diff
@@ -79,28 +117,20 @@ Users are supposed to provide 3 env vars for memory profiling.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The output files will be at the path where we call "ray start".
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:122]:
```suggestion
The output files are at the path where you call "ray start".
```

File: doc/source/ray-contribute/profiling.rst [hunk 13]
```diff
@@ -79,28 +117,20 @@ Users are supposed to provide 3 env vars for memory profiling.
...
+A example of profile file is "jeprof.15786.0.f.heap", then we can use following commands to generate the .svg plot.
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:123]:
```suggestion
For the example of profile file "jeprof.15786.0.f.heap", use following commands to generate the .svg plot.
```

File: doc/source/ray-contribute/profiling.rst [hunk 14]
```diff
@@ -79,28 +117,20 @@ Users are supposed to provide 3 env vars for memory profiling.
...
-  # If you realize the call graph is too large, use -focus=<some function> to zoom
-  # into subtrees.
-  google-pprof -focus=epoll_wait -svg $RAYLET /tmp/pprof.out > /tmp/pprof.svg
-
-Here's a snapshot of an example svg output, taken from the official
-documentation:
 
-.. image:: http://goog-perftools.sourceforge.net/doc/pprof-test-big.gif
 
 Running Microbenchmarks
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:135]:
```suggestion
Running microbenchmarks
```

File: doc/source/ray-contribute/profiling.rst [hunk 15]
```diff
@@ -118,4 +148,4 @@ References
 
 - The `pprof documentation <http://goog-perftools.sourceforge.net/doc/cpu_profiler.html>`_.
 - A `Go version of pprof <https://github.com/google/pprof>`_.
-- The `gperftools <https://github.com/gperftools/gperftools>`_, including libprofiler, tcmalloc, and other goodies.
+- The `gperftools <https://github.com/gperftools/gperftools>`_, including libprofiler, tcmalloc, and other goodies.
```
Comment (angelinalg) [inline doc/source/ray-contribute/profiling.rst:151]:
```suggestion
- The `gperftools <https://github.com/gperftools/gperftools>`_, including libprofiler, tcmalloc, and other useful tools.
```

**Review by angelinalg — APPROVED** [review]
Just some style nits and a typo.

**Pull request merged**